brokers/che-plugin-broker/broker/broker.go

```diff
@@ -114,24 +115,34 @@ func (cheBroker *ChePluginBroker) processPlugin(meta model.PluginMeta) error {
 	if err != nil {
 		return err
 	}
+	var pluginPath string
 
-	archivePath := filepath.Join(workDir, "pluginArchive.tar.gz")
-	pluginPath := filepath.Join(workDir, "plugin")
-
-	// Download an archive
-	cheBroker.PrintDebug("Downloading archive '%s' for plugin '%s:%s' to '%s'", url, meta.ID, meta.Version, archivePath)
-	err = cheBroker.ioUtil.Download(url, archivePath)
-	if err != nil {
-		return err
-	}
+	if strings.HasSuffix(url, pluginFileName) {
+		chePluginYamlPath := filepath.Join(workDir, pluginFileName)
+		cheBroker.PrintDebug("Downloading plugin definition '%s' for plugin '%s:%s' to '%s'", url, meta.ID, meta.Version, chePluginYamlPath)
+		err = cheBroker.ioUtil.Download(url, chePluginYamlPath)
+		if err != nil {
+			return err
+		}
+		pluginPath = workDir
+	} else {
+		archivePath := filepath.Join(workDir, "pluginArchive.tar.gz")
```

Comment:
I tried to follow the clean code idea for the brokers. I would rather see only one level of abstraction in this function.

Reply:
or one level of details if you will

Reply:
Ok. if I do two methods that get content. One will work with archive other with raw yaml. Will it work for you?

Reply:
Yeah, something like:
```go
switch (getTypeOfURL()) {
  case "archive":
    processArchive()
  case "yaml":
    processYAML()
}
```
seems clean for me. WDYT?

Reply:
easy

Reply:
@garagatyi can you take a look is this what you are asked for?